### DISCUSSION TITLE
How to have an admin user?

### DISCUSSION THREAD

**Discussion by phillipmohr** [supabase/supabase#7726] [Questions] [answered]
Hello everyone, I have a public website where users can register, but I also have an admin dashboard. I already managed that public users can register and login, but I don't know how I can distunguish between a public user and an admin user? 

It should only be possible to register admin users from the admin dashboard. It has to be secure. What would be the best way to do that? I'm using Nuxt 3, if that matters.

Thanks for any help!

**Comment by GaryAustin1** [comment, marked as answer]
This approach generated by a SB employee probably covers what you need: https://github.com/supabase-community/supabase-custom-claims
The nice thing is it uses the jwt for the claim so RLS is fast.

You can implement the same thing with an role or admin table and allow inserts/updates only by admin and then in RLS use select or a function to see if the user id is in the admin table.
